app/src/components/BoardsList/BoardsList.tsx

```diff
@@ -1,45 +1,40 @@
-import { useState } from "react";
-import { BoardItem } from "./BoardsList.types";
-import "@components/BoardsList/BoardsList.scss";
+import { useState, useContext } from "react";
+import styles from "@components/BoardsList/BoardsList.module.scss";
 import BoardIcon from "@assets/icons/board.svg?react";
 import PlusIcon from "@assets/icons/plus.svg?react";
-import Button from "@components/ui/Button/Button";
+import Button from "@components/ui/Button/Button/Button";
+import BoardModalContext from "@context/BoardModal/BoardModal";
+import BoardsContentContext from "@context/BoardsContent/BoardsContent";
 
 const BoardsList = () => {
-  const [boards, setBoards] = useState<BoardItem[]>([]);
-  const [activeElementId, setActiveElementId] = useState<number>();
+  const { boardsContent: boards } = useContext(BoardsContentContext);
```

Comment:
because you store here array of objects this may lead into many rerenders - context should be used mostly for primitives + things which change rarely. I saw this: https://react.dev/learn/scaling-up-with-reducer-and-context so combination of context + reducer - please check if it's worth to use this approach (so context is divided into two parts - one for boards, second for actions for it - dispatch)